### PR TITLE
fix dmlchannel with offline nodes

### PR DESCRIPTION
**Pull request by weiliu1031**
Signed-off-by: Wei Liu <wei.liu@zilliz.com>
Issue: https://github.com/milvus-io/milvus/issues/18652

### CODE REVIEW

**Review by congqixia — COMMENTED** [review]
will any offline nodes dml channel mapping missed if we just overwrite the watchDmlChannels node list?

File: internal/querycoord/cluster.go [hunk 1]
```diff
@@ -300,7 +300,16 @@ func (c *queryNodeCluster) WatchDmChannels(ctx context.Context, nodeID int64, in
...
 			old, ok := c.clusterMeta.getDmChannel(info.ChannelName)
 			if ok {
-				nodes = append(nodes, old.NodeIds...)
+				for _, oldNode := range old.NodeIds {
+					if ok, _ := c.IsOnline(oldNode); ok {
+						nodes = append(nodes, oldNode)
+					} else {
+						log.Debug("Remove offline node from dmChannel",
+							zap.String("channel", info.ChannelName),
+							zap.Int64("removed old node", oldNode),
```
Comment (yah01) [inline internal/querycoord/cluster.go:309]: log nameing convention: removedNode

**Comment by weiliu1031** [comment]
> will any offline nodes dml channel mapping missed if we just overwrite the watchDmlChannels node list?

aggree! we should just remove the source offline nodes.

File: internal/querycoord/cluster.go [hunk 2]
```diff
@@ -300,7 +300,12 @@ func (c *queryNodeCluster) WatchDmChannels(ctx context.Context, nodeID int64, in
 
 			old, ok := c.clusterMeta.getDmChannel(info.ChannelName)
 			if ok {
+				removeFromSlice(old.NodeIds, in.OfflineNodeID)
```
Comment (jaime0815) [inline internal/querycoord/cluster.go:303]: dmChannelInfos should be upated within cluster meta after triggering load balance of node down type. it seems not reasonable that remove offline node ids from a temporarily variable here.

Comment (weiliu1031) [reply internal/querycoord/cluster.go]: update logic problem: this pr means to remove offline node in time, when append new online node to dmChannel's nodes list.

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**